sql/sql_table.cc

```diff
@@ -16012,6 +16012,14 @@ static bool simple_rename_or_index_change(
     // It's now safe to take the table level lock.
     if (lock_tables(thd, table_list, alter_ctx->tables_opened, 0)) return true;
 
+#ifdef WITH_WSREP
+    /*
+      Locks are acquired for ALTER TABLE .. ENABLE | DISABLE KEYS
+      It is time to call nbo_phase_one_end().
+    */
+    WSREP_NBO_1ST_PHASE_END;
```

Comment:
Won't we hit WSREP_NBO_1ST_PHASE_END twice?

`alter table t2 enable keys, rename to t1;`

Reply:
Hi! Yes. It is possible that it calls `WSREP_NBO_1ST_PHASE_END` twice for query like `alter table t1 rename to t2, enable keys;`

But the first call to `nbo_end_phase_one()` changes the toi_mode to `m_undefined`.

So the second call to `WSREP_NBO_1ST_PHASE_END` just becomes a no-op, as the check if `(!wrep_thd_is_local_nbo(thd)) return;` in `wsrep_NBO_end_phase_one()`  performs an early return and we are safe.